src/components/controller/controller.ts

```diff
@@ -1,22 +1,26 @@
+import { Data } from '../view/appView';
 import AppLoader from './appLoader';
 
 class AppController extends AppLoader {
-    getSources(callback) {
+    getSources(callback: (data?: Data) => void) {
         super.getResp(
             {
                 endpoint: 'sources',
             },
-            callback
+            callback,
         );
     }
 
-    getNews(e, callback) {
-        let target = e.target;
-        const newsContainer = e.currentTarget;
+    getNews(e: Event, callback: (data?: Data) => void) {
+        let target = e.target as HTMLElement;
```

Comment:
```suggestion
       let target: HTMLElement | null = <HTMLElement>e.target;
```
I am seeing only ```as HTMLElement``` above, but more often we use another case like this.
I recommend you to try it not only here but also in the code below

Reply:
All the 'as' cases are fixed now